pkg/imagebuilder/content.go

```diff
@@ -119,13 +141,42 @@ func (c ConfigBuilder) getUnits() ([]igntypes.Unit, error) {
 			return units, fmt.Errorf("Failed to read unit %s: %w", e.Name(), err)
 		}
 
+		templated, err := templateString(e.Name(), string(contents))
+		if err != nil {
+			return units, err
+		}
+
 		unit := igntypes.Unit{
 			Name:     e.Name(),
 			Enabled:  ignutil.BoolToPtr(true),
-			Contents: ignutil.StrToPtr(string(contents)),
+			Contents: ignutil.StrToPtr(string(templated)),
 		}
 		units = append(units, unit)
 	}
 
 	return units, nil
 }
+
+func templateString(name string, text string) (string, error) {
+	params := map[string]interface{}{
+		// TODO: try setting SERVICE_BASE_URL within agent.service
+		"ServiceBaseURL": os.Getenv("SERVICE_BASE_URL"),
+		// TODO: get id either from InfraEnv CR that is included
+		// with tool, or query the id from the REST_API
+		// curl http://SERVICE_BASE_URL/api/assisted-install/v2/infra-envs
+		"infraEnvId": os.Getenv("INFRA_ENV_ID"),
+		// TODO: needs appropriate value if AUTH_TYPE != none
+		"PullSecretToken": os.Getenv("PULL_SECRET_TOKEN"),
+	}
```

Comment:
I had imagined any template params we need being passed in the ConfigBuilder struct, but this is fine for now.

Reply:
Ditto